src/Faithlife.Reflection/TupleInfo.cs

```diff
@@ -81,6 +82,7 @@ public sealed class TupleInfo<T> : ITupleInfo
 		/// <summary>
 		/// Creates a tuple from the specified items.
 		/// </summary>
+		[return: NotNull]
```

Comment:
It seems like this should be unnecessary because `T` is not a nullable type? That is, doesn't the method signature already convey this?

Reply:
IIRC, I was able to remove exclamation points after `CreateNew` calls after adding this attribute.